pliers/extractors/text.py

```diff
@@ -402,3 +400,140 @@ def _extract(self, stim):
+            (PyTorch) or 'tensorflow'. Defaults to 'pt'.
+        encoding_level (str): A string specifying whether encodings for each
+            token or sequence encodings are to be returned. Must be one of
+            'token', 'sequence'.
+        pooling (str): Optional argument, relevant for sequence-level
+            embeddings only. If None and encoding_level='sequence', encodings
+            for [CLS] tokens are returned. If encoding_level='sequence' and
+            numpy function is specified, token-level embeddings are pooled
+            according to specified method (e.g. 'mean', 'max', 'min').
+        model_kwargs (dict): Named arguments for pretrained model.
+            See: https://huggingface.co/transformers/main_classes/model.html
+            and https://huggingface.co/transformers/model_doc/bert.html for
+            further info.
+        tokenizer_kwargs (dict): Named arguments for tokenizer.
+            See https://huggingface.co/transformers/main_classes/tokenizer.html
+            for further info.
+    '''
+
+    _log_attributes = ('pretrained_model', 'encoding_level',
+                       'pooling', 'framework', 'tokenizer_type',
+                       'model_kwargs', 'tokenizer_kwargs')
+
+    def __init__(self,
+                 pretrained_model_or_path='bert-base-uncased',
+                 tokenizer='bert-base-uncased',
+                 framework='pt',
+                 encoding_level='token',
+                 pooling=None,
+                 model_kwargs=None,
+                 tokenizer_kwargs=None):
+                 
+        verify_dependencies(['transformers'])
+
+        if framework not in ['pt', 'tf']:
+            raise(ValueError('''Invalid framework;
+                must be one of 'pt' (pytorch) or 'tf' (tensorflow)'''))
+
+        if model_kwargs is None:
+            model_kwargs = {}
+
+        if tokenizer_kwargs is None:
+            tokenizer_kwargs = {}
+
+        self.pretrained_model = pretrained_model_or_path
+        self.tokenizer_type = tokenizer
+        self.framework = framework
+        self.encoding_level = encoding_level
+        self.pooling = pooling
+        self.model_kwargs = model_kwargs
+        self.tokenizer_kwargs = tokenizer_kwargs
+        model_name = 'BertModel' if self.framework == 'pt' else 'TFBertModel'
+
+        self.tokenizer = transformers.BertTokenizer.from_pretrained(
+            tokenizer, **tokenizer_kwargs)
+        self.model = getattr(transformers, model_name).from_pretrained(
+            pretrained_model_or_path, **model_kwargs)
+
+        super(PretrainedBertEncodingExtractor, self).__init__()
+
+    def _extract(self, stims):
+
+        stims.name = ' '.join([s.text for s in stims.elements])
+        text, onsets, durations = zip(
+            *((s.text, s.onset, s.duration) for s in stims.elements))
+        durations = [0.0 if d is None else d for d in durations]
```

Comment:
Not sure if we should default to `0` here or `None`. I believe if it's `None` for all values, then the `duration` column would be dropped from the `to_df` call, which might be desirable behavior. Even if that's not true, `0` is different from no value. Is there a reason we need to set this to `0`, as opposed to keeping `None`?

Reply:
the reason was that the logic for the `cast_to_token_level` function included repeating elements using * operator which does not support None objects. Changed the function, replaced * with `itertools.repeat`. Should be fixed now.